addOns/llm/src/main/java/org/zaproxy/addon/llm/services/LlmCommunicationService.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2025 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.llm.services;
+
+import com.azure.ai.openai.models.ChatCompletionsJsonResponseFormat;
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.MessageWindowChatMemory;
+import dev.langchain4j.model.azure.AzureOpenAiChatModel;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.ollama.OllamaChatModel;
+import dev.langchain4j.service.AiServices;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.httpclient.util.HttpURLConnection;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpSender;
+import org.zaproxy.addon.llm.LlmOptions;
+import org.zaproxy.addon.llm.communication.Confidence;
+import org.zaproxy.addon.llm.communication.HttpRequestList;
+import org.zaproxy.addon.llm.utils.HistoryPersister;
+import org.zaproxy.addon.llm.utils.Requestor;
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+
+public class LlmCommunicationService {
+
+    private static final Logger LOGGER = LogManager.getLogger(LlmCommunicationService.class);
+    private static final String AI_REVIEWD_TAG_KEY = "AI-Reviewed";
```

Comment:
"AI_REVIEWD_TAG_KEY" --> "AI_REVIEWED_TAG_KEY"

Reply:
Oops yeah it's missing an "E"